src/ruby/puppetserver-lib/puppet/server/master.rb

```diff
@@ -57,6 +57,7 @@ def initialize(puppet_config, puppet_server_config)
           "supports threadsafe settings. Please upgrade to Puppet 6.13.0 or greater.")
       end
     end
+    check_cadir_for_deprecation_warning
```

Comment:
So doing this here, this will run on every JRuby recycle. Is that what we want, as opposed to logging this in Clojure, say when the certificate authority starts?

Reply:
I feel like since the setting being deprecated is a Puppet setting, it made more sense to run the message from Puppet itself. I don't have a strong opinion either way, but it seemed weirder to me to warn in clojure about a setting that Puppet itself is removing.